main/acle.md

```diff
@@ -4726,6 +4881,2404 @@ assertions, which may speed up access to aligned data (and will fault
+    operate on VLAT. For example:
+
+    ``` c
+      #if __ARM_FEATURE_SVE_BITS==512
+      typedef svbool_t pred __attribute__((arm_sve_vector_bits(512)));
+      pred f(pred x, pred y) { return svbrka_z(x, y); }
+      #endif
+    ```
+
+*   A conditional expression of the form `C ? E1 : E2` is ill-formed if:
+
+    *   E1 has type VLAT or cv-qualified VLAT and E2 has type VLST or
+        cv-qualified VLST; or
+
+    *   E1 has type VLST or cv-qualified VLST and E2 has type VLAT or
+        cv-qualified VLAT.
+
+    However, the implementation does not need to diagnose these cases.
+
+    This rule avoids any ambiguity about whether the result of the
+    conditional expression is sized or sizeless. For example:
+
+    ``` c
+      #if __ARM_FEATURE_SVE_BITS==512
+      typedef svint32_t vec __attribute__((arm_sve_vector_bits(512)));
+      auto f(bool cond, vec x, svint32_t y) { return cond ? x : y; }
+      #endif
+    ```
+
+    is ill-formed because it is not clear whether the type of the result
+    should come from `x` or `y`.
+
+*   VLST maps to the same AAPCS64 ABI type as VLAT. For example:
+
+    ``` c
+      svint32_t __attribute__((arm_sve_vector_bits(128)))
+    ```
+
+    maps to the same AAPCS64 type as `svint32_t`, rather than to the
+    same AAPCS64 type as the `arm_neon.h` type `int32x4_t`.
+
+*   An object type “requires size N” if:
+
+    *   it is an SVE vector type or SVE predicate type that has the
+        attribute `arm_sve_vector_bits(N)`;
+
+    *   it is an array type whose element type requires size N; or
+
+    *   it is a class or struct and:
+
+        *   at least one non-static member has a type that requires size N; or
+
+        *   a base type requires size N
+
+    Note that this definition specifically excludes reference and pointer
+    types.
+
+*   A function type “requires size N” if the return type requires size N
+    or if at least one parameter type requires size N.
+
+*   A program is ill-formed if:
+
+    *   any function type requires two different sizes N1 and N2.
+
+    *   `__ARM_FEATURE_SVE_BITS!=N` when defining a function whose type
+        requires size N.
+
+    *   `__ARM_FEATURE_SVE_BITS!=N` when calling a function whose type
+        requires size N. (Note that this explicitly excludes parameters
+        passed through “`...`”.)
+
+    *   an object whose type requires some size N is passed to an
+        unprototyped function.
+
+    However, the implementation does not need to diagnose these cases.
+
+    These rules only have an effect if an implementation allows the
+    register size to vary within a translation unit. The aim is to forbid
+    function definitions and function calls that have no mapping to the
+    AAPCS64 calling conventions. For example, if an implementation
+    supports some form of pragma that changes `__ARM_FEATURE_SVE_BITS`
+    within a translation unit:
+
+    ``` c
+      …pragma that changes __ARM_FEATURE_SVE_BITS to 256…
```

Comment:
I've replaced `…` with `...` throughout (in code and outside).  I didn't comment the lines because I wanted to make it clear that this was supposed to be a stand-in for real code.